File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.swift
import Foundation

let input = Int(readLine()!)!

var f: [Int] = .init(repeating: 0, count: input + 1)

func fibonacci(n: Int) -> Int {
    f[1] = 1
    f[2] = 1
    for i in 3...n {
        f[i] = f[i - 1] + f[i - 2]
    }
    return f[n];
}

print(fibonacci(n: input), input - 2)

File: 백준/Bronze/24416. 알고리즘 수업 － 피보나치 수 1/알고리즘 수업 － 피보나치 수 1.py
a = int(input())
def fib(n):
    count = 0
    dp = [1] * n
    if n < 2: return 1
    for i in range(2,n):
        dp[i] = dp[i-1] + dp[i-2]
        count += 1
    return dp[n-1], count

fib, count = fib(a)
print(fib, count)